File: palindrome/palindrome.js
function decrementChar(c) {
	return String.fromCharCode(c.charCodeAt(0) - 1);
}

function incrementChar(c) {
	return String.fromCharCode(c.charCodeAt(0) + 1);
}

function makePalindrome(input) {
	let steps = 0;
	for (let i = 0; i < input.length / 2; i++) {
		let firstLetter = input[i];
		let lastLetter = input[input.length - i - 1];
		while (firstLetter !== lastLetter) {
			steps++;
			if (firstLetter.localeCompare(lastLetter) === -1) {
				lastLetter = decrementChar(lastLetter);
			} else {
				lastLetter = incrementChar(lastLetter);
			}
		}
	}
	return steps;
}

console.log(`abc > ${makePalindrome("abc")} steps`);
console.log(`abcde > ${makePalindrome("abcde")} steps`);
console.log(`aaa > ${makePalindrome("aaa")} steps`);

// Run with `node palindrome.js`

File: palindrome/palindrome.go
package palindrome

import (
	"errors"
	"regexp"
	"strings"
)

const alphabets string = "abcdefghijklmnopqrstuvwxyz"

func intToAlphabet(i int) string {
	return string('a' + i)
}

func alphabetToInt(alphabet string) int {
	return strings.Index(alphabets, alphabet)
}

func MakePalindrome(input string) (int, error) {
	match, _ := regexp.MatchString("^[a-z]*$", input)
	if !match {
		return -1, errors.New("can only work with lowercase alphabet")
	}
	steps := 0
	for i := 0; i < len(input)/2; i++ {
		firstLetter, lastLetter := string(input[i]), string(input[len(input)-i-1])
		for firstLetter != lastLetter {
			steps++
			if strings.Compare(firstLetter, lastLetter) == -1 {
				lastLetter = intToAlphabet(alphabetToInt(lastLetter) - 1)
			} else {
				lastLetter = intToAlphabet(alphabetToInt(lastLetter) + 1)
			}
		}
	}
	return steps, nil
}
